files/__addonLocation__/rollup.config.__rollupExt__

```diff
@@ -14,14 +14,20 @@ export default {
   plugins: [
     // These are the modules that users should be able to import from your
     // addon. Anything not listed here may get optimized away.
-    addon.publicEntrypoints(['components/**/*.js', 'index.js']),
+    addon.publicEntrypoints(['components/**/*.<%= ext %>', 'index.<%= ext %>'<% if (typescript) {%>, 'glint.ts'<% } %>]),
 
     // These are the modules that should get reexported into the traditional
     // "app" tree. Things in here should also be in publicEntrypoints above, but
     // not everything in publicEntrypoints necessarily needs to go here.
     addon.appReexports(['components/**/*.js']),
 
-    // This babel config should *not* apply presets or compile away ES modules.
+<% if (typescript) { %>    // compile TypeScript to latest JavaScript, including Babel transpilation
+    typescript({
+      transpiler: 'babel',
+      browserslist: false,
```

Comment:
Yeah, I remember this discussion in your other PR. As I said in the PR description, I just added what worked for me, *for now*. When your PR is merged, I will update all of this! So let's not duplicate the same discussion here! :wink:

Reply:
For the record: as I already stated in the related https://github.com/embroider-build/embroider/pull/1099, I think the config here is right (`browserslist: false`) as documented by `rollup-plugin-ts`, it's just that the latter seems to not behave correctly here. See the issue I filed here: https://github.com/wessberg/rollup-plugin-ts/issues/189

Given that this does not break anything, it just produces code that is not as optimized as it could be, I would think to keep this as is now, and not expose workarounds to created addons that later need to get rolled back causing churn for maintainers, and rather rely on getting this fixed upstream eventually!